docs/source/model/dependencies.rst

```diff
@@ -0,0 +1,453 @@
+
+  .. code-block:: yaml
+
+      python: 3.9.8
+      build_dependencies:
+      - pip==23.3.2
+      - setuptools==69.0.3
+      - wheel==0.42.0
+      dependencies:
+      - -r requirements.txt
+
+* ``requirements.txt``
+
+  .. code-block:: text
+
+      mlflow==2.9.2
+      scikit-learn==1.3.2
+      cloudpickle==3.0.0
+
+* ``conda.yaml``
+
+  .. code-block:: yaml
+
+      name: mlflow-env
+      channels:
+        - conda-forge
+      dependencies:
+      - python=3.9.8
+      - pip
+      - pip:
+        - mlflow==2.9.2
+        - scikit-learn==1.3.2
+        - cloudpickle==3.0.0
+
+
+Adding Extra Dependencies to MLflow Model
+-----------------------------------------
+MLflow detects dependencies required for the model flavor library, but your model may depend on other libraries e.g. data
+preprocessing. In this case, you can add extra dependencies to the model by specifying the **extra_pip_requirements** param
+when logging the model. For example,
+
+.. code-block:: python
+
+    import mlflow
+
+
+    class CustomModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            # your model depends on pandas
+            import pandas as pd
+
+            ...
+            return prediction
+
+
+    # Log the model
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model(
+            python_model=CustomModel(),
+            artifact_path="model",
+            extra_pip_requirements=["pandas==2.0.3"],
+        )
+
+The extra dependencies will be added to ``requirements.txt`` as follows (and similarly to ``conda.yaml``):
+
+.. code-block:: yaml
+
+    mlflow==2.9.2
+    cloudpickle==3.0.0
+    pandas==2.0.3  # added
+
+
+Then MLflow will install Pandas 2.0.3 when serving the model for prediction.
+
+.. note::
+
+    Once you log the model with dependencies, it is advisable to test prediction in a sandbox environment using MLflow predict API,
+    to avoid any dependency issues when deploying the model to production.
+    Please refer to :ref:`Validating Environment for Prediction <validating-environment-for-prediction>` for more details.
+
+Defining All Dependencies by Yourself
+--------------------------------------
+
+Alternatively, you can also define all dependencies from scratch rather than adding extra ones. To do so,
+specify **pip_requirements** when logging the model. For example,
+
+.. code-block:: python
+
+    import mlflow
+
+    # Log the model
+    with mlflow.start_run() as run:
+        mlflow.sklearn.log_model(
+            sk_model=model,
+            artifact_path="model",
+            extra_pip_requirements=[
+                "mlflow-skinny==2.9.2",
+                "cloudpickle==2.5.8",
+                "sklearn==1.3.1",
+            ],
+        )
+
+The manually defined dependencies will override the default ones MLflow detects from the model flavor library:
+
+.. code-block:: yaml
+
+    mlflow-skinny==2.9.2
+    cloudpickle==2.5.8
+    sklearn==1.3.1
+
+.. note::
```

Comment:
Let's add another warning here that manually declaring core dependencies for the model to function that are different from those that were used during training can be especially dangerous and prone to unexpected behavior.